mmengine/utils/progressbar_rich.py

```diff
@@ -0,0 +1,330 @@
+                    f'{self.descriptions[task_id]}'
+                    f'_{completed}/{total}')
+
+                if self._is_task_finish(task_id):
+                    self.bar.stop()
+            else:
+                self.completed += advance
+                elapsed = self.timer.since_start()
+                if elapsed > 0:
+                    fps = self.completed / elapsed
+                else:
+                    fps = float('inf')
+                self.write(
+                    f'completed: {self.completed}, '
+                    f'elapsed: {int(elapsed + 0.5)}s, '
+                    f'{fps:.1f} tasks/s', self.colors[task_id])
+        else:
+            assert task_id >= 0 and task_id < len(self.tasks)
+            completed = self.bar.tasks[task_id].completed + 1
+            total = self.bar.tasks[task_id].total
+            self.bar.update(
+                task_id,
+                advance=advance,
+                description=f'[{self.colors[task_id]}]'
+                f'{self.descriptions[task_id]}'
+                f'_{completed}/{total}')
+
+            if self._finished():
+                self.bar.stop()
+
+
+def track_single_progress(func,
+                          tasks,
+                          task_num=None,
+                          description='Process...',
+                          color='blue',
+                          **kwargs):
+    """Track the progress of tasks execution with a progress bar.
+
+    Tasks are done with a simple for-loop.
+
+    Args:
+        func (callable): The function to be applied to each task.
+        tasks (list or tuple[Iterable, int]): A list of tasks or
+            (tasks, total num).
+        description (str): The description of progress bar.
+        color (str): The color of progress bar.
+
+    Returns:
+        list: The task results.
+    """
+    if isinstance(tasks, tuple):
+        assert len(tasks) == 2
+        assert isinstance(tasks[0], Iterable)
+        assert isinstance(tasks[1], int)
+        tasks = tasks[0]
+        if task_num is not None:
+            assert task_num == tasks[1]
+    elif isinstance(tasks, Iterable):
+        if task_num is not None:
+            assert task_num == len(tasks)
+    else:
+        raise TypeError(
+            '"tasks" must be an iterable object or a (iterator, int) tuple')
+    prog_bar = RichProgressBar()
+    prog_bar.add_single_task(task_num, color=color, description=description)
+    results = []
+    for task in tasks:
+        results.append(func(task, **kwargs))
+        prog_bar.update()
+    return results
+
+
+def init_pool(process_num, initializer=None, initargs=None):
+    if initializer is None:
+        return Pool(process_num)
+    elif initargs is None:
+        return Pool(process_num, initializer)
+    else:
+        if not isinstance(initargs, tuple):
+            raise TypeError('"initargs" must be a tuple')
+        return Pool(process_num, initializer, initargs)
```

Comment:
Why do we need to define `init_pool` here? Can we directly call `Pool(xxx)` to create it?